docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,384 @@
+
+### Step 3: Persist DesktopAgentDetails to SessionStorage
+
+Once a connection is established, and the app and instance identity determined, a `DesktopAgentDetails` record must be stored in SessionStorage under the `FDC3-Desktop-Agent-Details` key by the `getAgent()` implementation. This record includes:
+
+- The `identityUrl` and `actualUrl` passed to `getAgent()`.
+- The `appId`, `instanceId`, and `instanceUuid` assigned by the DA.
+- An `agentUrl` field with the URL provided in any [`WCP2LoadUrl`](https://fdc3.finos.org/schemas/next/api/WCP2LoadUrl.schema.json) message that was received
+  - Used to skip sub-steps 1-3 in the discovery process described in Step 1.2.
+  - If no [`WCP2LoadUrl`](https://fdc3.finos.org/schemas/next/api/WCP2LoadUrl.schema.json) message was received, omit this field.
+- An `agentType` field which indicates what type of connection to the DA was established
+  - Used to limit the discovery process in Step 1.2 above to only allow agents of the same type.
+  - the `getAgent()` implementation should determine what value to set, from the `WebDesktopAgentType` enumeration for this field based on what happened during the discovery process.
+
+See the [reference documentation for getAgent()](../ref/GetAgent#persisted-connection-data) for further details of the `DesktopAgentDetails` record that should be saved to SessionStorage.
+
+:::tip
+
+SessionStorage is ephemeral, only existing for the duration of the window's life. Hence, there is no concern with the key being overwritten or conflicting with other DAs.
+
+:::
+
+### Step 4: Resolve promise with DesktopAgent interface (step 4)
+
+Resolve the `getAgent()` promise with an object containing either a `DesktopAgent` implementation (that was found at `window.fdc3` or returned by a `failover` function) or a 'Desktop Agent Proxy' implementation (a class implementing the `DesktopAgent` interface that uses the [Desktop Agent Communication Protocol (DACP)](./desktopAgentCommunicationProtocol) to communicate with a Desktop Agent over the `MessagePort`).
+
+Where a `DesktopAgent` or 'Desktop Agent Proxy' implementation was successfully returned, any subsequent calls to `getAgent()` that are not preceded by a navigation or refresh event, should resolve to the same instance.
+
+### Step 5: Disconnection
+
+Desktop Agent Preload interfaces, as used in container-based Desktop Agent implementations, are usually able to track the lifecycle and current URL of windows that host web apps in their scope. Hence, this is currently no requirement nor means for an app to indicate that it is closing, rather it is the responsibility of the Desktop Agent to update its internal state when an app closes or changes identity.
```

Comment:
"this is currently no requirement" should probably be "there is currently no requirement"

Reply:
Fixed in 5e8ba8b01ab5ecf86081172ff87369b1faf28540